GibbsSampling.Library/MotifSampler/MotifSamplerFunctions.fs

```diff
@@ -0,0 +1,334 @@
+﻿namespace MotifSampler
+
+open FSharpAux
+open BioFSharp
+open CompositeVector.Types
+open CompositeVector.Functions
+open PositionMatrix.Types
+open PositionMatrix.Functions
+open SiteSampler.Functions
+open MotifSampler.Types
+
+module Functions =
+    
+    /// Creates a MotifIndex based on the probability of the given segement(s) start position(s).
+    let createMotifIndex pwms pos =
+        {
+            MotifIndex.PWMS         = pwms
+            MotifIndex.Positions    = pos
+        }
+
+    /// Calculates all possible segment combination of m segments, which do not overlap in the given width 
+    /// and gives you the probability for it.
+    let calculatePWMsForSegmentCombinations (cutOff:float) (width:int) (m:int) (set:list<float*int>) =
+        let rec loop prob positions size set = 
```

Comment:
readability can be improved here:

- As far as i understand this function it can only ever return a sequence with a single element or none element. If this is true, remove all sequence logic and return a option, on which you can use List.singleton or List.empty in the end.
- Use guard cases in your pattern matching to reduce if-logic tree
    `| n, x::xs when n > 0`

Reply:
Issue erstellt: https://github.com/Etschbeijer/GibbsSampling2.0/issues/10